examples/microwave-oven-app/microwave-oven-common/include/microwave-oven-control-delegate.h

```diff
@@ -0,0 +1,83 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/microwave-oven-control-server/microwave-oven-control-server.h>
+#include <app/util/af-enums.h>
+#include <functional>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace MicrowaveOvenControl {
+
+// This is an application level delegate to handle microwave oven control commands according to the specific
+// business logic.
+class ExampleMicrowaveOvenControlDelegate : public MicrowaveOvenControl::Delegate
+{
+
+private:
+    std::function<Protocols::InteractionModel::Status(uint8_t, uint32_t, uint8_t)> mHandleSetCookingParametersCallback;
```

Comment:
is this intended to be used as a baseline example for embedded devices or be common? std::function uses heap, so we generally do not use it in embedded.

Reply:
This is an implementation example for the embeded devices.

Would it help if uses unique_ptr upon all those callback function pointers? 
such as ` std::unique_ptr<std::function<void(Clusters::OperationalState::GenericOperationalError & err)>> mPauseCallback;`
In this way, there would be no memery problem when the program has unexpected shutdown.




> is this intended to be used as a baseline example for embedded devices or be common? std::function uses heap, so we generally do not use it in embedded.